### PR TITLE
[GLUTEN-6656][UNIFFLE] VeloxUniffleColumnarShuffleWriter should send commit for all ColumnBatch with empty rows

### PR DESCRIPTION
**Pull request by SteNicholas**
## What changes were proposed in this pull request?

When all of the ColumnarBatch have empty rows in `VeloxUniffleColumnarShuffleWriter`, the `nativeShuffleWriter` is not initialized and causes `IllegalStateException`. `VeloxUniffleColumnarShuffleWriter` should send commit for all `ColumnBatch` with empty rows.

Follow up #6657.

(Fixes: \#6656)

## How was this patch tested?

CI.

### CODE REVIEW

**Comment by SteNicholas** [comment]
Ping @exmy, @jackylee-ch, @summaryzb.

File: gluten-uniffle/velox/src/main/java/org/apache/spark/shuffle/writer/VeloxUniffleColumnarShuffleWriter.java [hunk 1]
```diff
@@ -189,11 +189,13 @@ public long spill(MemoryTarget self, Spiller.Phase phase, long size) {
...
 
-    long startTime = System.nanoTime();
     LOG.info("nativeShuffleWriter value {}", nativeShuffleWriter);
-    if (nativeShuffleWriter == -1L) {
-      throw new IllegalStateException("nativeShuffleWriter should not be -1L");
+    // If all of the ColumnarBatch have empty rows, the nativeShuffleWriter still equals -1
+    if (nativeShuffleWriter == -1L && !isMemoryShuffleEnabled) {
+      super.sendCommit();
+      return;
     }
```
Comment (xianjingfeng) [inline gluten-uniffle/velox/src/main/java/org/apache/spark/shuffle/writer/VeloxUniffleColumnarShuffleWriter.java:197]:
```suggestion
    if (nativeShuffleWriter == -1L) {
      if (!isMemoryShuffleEnabled) {
        super.sendCommit();
      }
      return;
    }
```

Comment (SteNicholas) [reply gluten-uniffle/velox/src/main/java/org/apache/spark/shuffle/writer/VeloxUniffleColumnarShuffleWriter.java:197]: @xianjingfeng, when `!records.hasNext()` returns true but `isMemoryShuffleEnabled` is true, the `VeloxUniffleColumnarShuffleWriter#writeImpl` does not also return. This change keeps the same behavitor with above.

**Comment by summaryzb** [comment]
I'm ok with this fix

File: gluten-uniffle/velox/src/main/java/org/apache/spark/shuffle/writer/VeloxUniffleColumnarShuffleWriter.java (hunk 1 above)
Comment (xianjingfeng) [reply gluten-uniffle/velox/src/main/java/org/apache/spark/shuffle/writer/VeloxUniffleColumnarShuffleWriter.java:197]: > @xianjingfeng, when `!records.hasNext()` returns true but `isMemoryShuffleEnabled` is true, the `VeloxUniffleColumnarShuffleWriter#writeImpl` does not also return. This change keeps the same behavior with above.

I don't think the above logic makes sense either.

Comment (xianjingfeng) [reply gluten-uniffle/velox/src/main/java/org/apache/spark/shuffle/writer/VeloxUniffleColumnarShuffleWriter.java:197]: https://github.com/apache/incubator-uniffle/blob/master/client-spark/spark3/src/main/java/org/apache/spark/shuffle/writer/RssShuffleWriter.java#L341-L343

**Review by xianjingfeng — APPROVED** [review]
LGTM

**Comment by SteNicholas** [comment]
@jackylee-ch, PTAL.

**Review by jackylee-ch — APPROVED** [review]
LGTM

**Pull request merged**